snuba/querylog/__init__.py

```diff
@@ -21,29 +30,47 @@ def _record_timer_metrics(
     request: Request,
     timer: Timer,
     query_metadata: SnubaQueryMetadata,
+    result: Union[QueryResult, QueryException, QueryPlanException],
 ) -> None:
     final = str(request.query.get_final())
     referrer = request.referrer or "none"
     app_id = request.attribution_info.app_id.key or "none"
     parent_api = request.attribution_info.parent_api or "none"
+    tags = {
+        "status": query_metadata.status.value,
+        "request_status": query_metadata.request_status.value,
+        "slo": query_metadata.slo.value,
+        "referrer": referrer,
+        "parent_api": parent_api,
+        "final": final,
+        "dataset": query_metadata.dataset,
+        "app_id": app_id,
+    }
+    mark_tags = {
+        "final": final,
+        "referrer": referrer,
+        "parent_api": parent_api,
+        "dataset": query_metadata.dataset,
+    }
+    if isinstance(result, QueryPlanException):
+        # The QueryPlanException is raised outside the query execution flow.
+        # As a result, its status and SLO values are not based on its query_list
+        if result.exception_type == StorageNotAvailable.__name__:
+            tags = {
+                "status": QueryStatus.ERROR.value,
```

Comment:
I would rather you used the `get_request_status` for the `request_status` and `slo` parameters. I am trying to keep the logic that converts exceptions -> statuses central in one place.

Reply:
Makes sense, updated.